components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -876,10 +876,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         const user = await this.checkAndBlockUser("getOwnerToken");
 
         //TODO this requests are only here to populate the resource guard check
-        const workspace = await this.workspaceService.getWorkspace(user.id, workspaceId);
-        if (!workspace) {
-            throw new Error("owner token not found");
-        }
+        const { workspace } = await this.workspaceService.getWorkspace(user.id, workspaceId);
         await this.guardAccess({ kind: "workspace", subject: workspace }, "get");
 
         const latestInstance = await this.workspaceService.getCurrentInstance(user.id, workspaceId);
```

Comment:
would make sense to use the instance that is returned by ws.getWorkspace ?

Reply:
In general I kept the pattern of "don't touch the old checks/code as much as possible, so it can be removed more easily (and I don't have to move to much code now).
Thus I did not go into:
 - replacing duplicate instance fetches (where in some cases "current"/"running" semantics have to be replicated)
 - removing the `check` function (they are just very convenient to throw away later)

I can totally go for the "more integrated" way, just did not think it's worth it...?

Reply:
(This place uses the NOT_FOUND check, for instance).